tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-indexed-access-negative.bal

```diff
@@ -19,3 +10,27 @@ function testNonStringIndexAccess() {
     var k = x[true];
     var b = x[1.2];
 }
+
+function testInvalidXMLAccessWithIndex() {
+    xml x1 = xml `<root><!-- comment node--><name>supun</name><city>colombo</city></root>`;
+    xml x2 = x1[0]/*;
+
+    x2[1] = xml `<address/>`;
+
+    string|'xml:Text x3 = xml `sample test`;
+    string sampleString1 = x3[0];
+    xml x4 = x3[0];
+
+    'xml:Element x5 = xml `<root><!-- comment node--><name>anne</name><city>colombo</city></root>`;
+    xml x6 = x5[0]/*;
```

Comment:
Thinking about this should not be an error,  (https://github.com/ballerina-platform/ballerina-spec/issues/727)
x5 -> xml<element> 
x5[0] -> xml<element>|xml<never> -> xml<element> 
hence x5[0]/* should be allowed.